qubes/storage/file.py

```diff
@@ -328,12 +344,15 @@ def start(self):
         return self
 
     def stop(self):
+        assert self._locked is FileVolume._marker_running, \
+            'cannot stop a volume that has not been started'
```

Comment:
This breaks if a VM is started, then `qubesd` is restarted (recreating a FileVolume object with `_locked == None`), then the VM is stopped.

Similar for `export_end()`, I suppose.

Reply:
Is `is_dirty` guaranteed to be `True` in that situation?

Reply:
Hmm maybe not quite guaranteed, e.g. if `qubesd` is restarted while Xen is still running the `block-origin` script (so that `st_blocks` is not yet `> 0`).

Reply:
In that case, VM wouldn't fully start - at best would remain "paused" (it's only unpaused by qubesd after fully constructing - if interrupted here, it won't unpause). If qubesd gets restarted at this stage, I don't think such VM will boot properly, even if gets unpaused manually later, for example qubesdb won't be there.
For all practical purposes, I think it is ok to assume `is_dirty` is true when VM is running.